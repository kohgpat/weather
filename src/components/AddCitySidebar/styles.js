import styled from "styled-components";
import ButtonComponent from "../Button";

export const Sidebar = styled.div`
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  width: 300px;
  background-color: #fff;
  display: none;
  z-index: 1;
  border-left: var(--add-city-sidebar-border);

  ${props => props.isVisible && `
    display: flex;
    flex-direction: column;
  `}
`;

export const SidebarHeader = styled.header`
  display: flex;
  padding: 20px;
`;

export const SidebarCloseButton = styled(ButtonComponent).attrs({ type: "button" })`
  width: auto;
  margin-left: auto;
`;
