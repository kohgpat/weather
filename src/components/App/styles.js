import styled from "styled-components";

export const App = styled.div`
  height: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
`;
