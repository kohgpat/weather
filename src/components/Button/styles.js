import styled from "styled-components";

export const Button = styled.button`
  font-size: 14px;
  font-weight: bold;
  font-family: "Roboto Condensed", "Helvetica Neue", "Helvetica", Arial, sans-serif;
  padding: 8px 16px;
  height: 40px;
  border: 0;
  background-color: #eee;
  width: 100%;
`;
