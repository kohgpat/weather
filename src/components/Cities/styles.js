import styled from "styled-components";

export const Cities = styled.div`
  margin-top: calc(50px + var(--topbar-height));
  padding: 20px;

  @supports (display: grid) {
    display: grid;
    grid-template-columns: repeat(1, 1fr);
    grid-gap: 30px;
    grid-auto-rows: minmax(300px, auto);

    @media (min-width: 768px) {
      grid-template-columns: repeat(2, 1fr);
    }

    @media (min-width: 1028px) {
      grid-template-columns: repeat(3, 1fr);
    }
  }

  @supports not (display: grid) {
    display: flex;
    flex-direction: row;
    flex-shrink: 0;
    align-items: center;
    justify-content: center;
    width: 100%;
    flex-wrap: wrap;
  }
`;

export const NoCitiesMessage = styled.div`
  margin-top: calc(50px + var(--topbar-height));
  padding: 20px;
`;
