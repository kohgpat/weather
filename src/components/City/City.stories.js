/* global module */
import React from "react";
import { storiesOf } from "@storybook/react";
import { action } from "@storybook/addon-actions";
import City from "./index";

const city = {
  name: "Moscow",
  weather: [{ main: 22, temp: 22 }]
};

const settings = {
  units: "celcius"
};

storiesOf("City", module).add("default", () => (
  <City city={city} settings={settings} onRemoveCity={action("onRemoveCity")} />
));
