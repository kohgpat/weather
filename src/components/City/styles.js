import styled from "styled-components";

export const City = styled.div`
  height: 280px;
  width: 280px;
  background-color: #fff;
  border-radius: 4px;
  box-shadow: 0 0 20px -2px rgba(0,0,0,0.1);
  display: flex;
  flex-direction: column;
  padding: 10px;

  @media (min-width: 768px) {
    height: 300px;
    width: 300px;
  }

  @supports not (display: grid) {
    margin-top: 30px;

    @media (orientation: landscape) {
      &:not(:last-child) {
        margin-right: 30px;
      }
    }

    @media (min-width: 768px) {
      &:not(:last-child) {
        margin-right: 30px;
      }
    }
  }
`;

export const Header = styled.header`
  display: flex;
  align-items: center;
  justify-content: space-between;
`;

export const Name = styled.h1`
  margin: 10px;
  width: 80%;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
`;

export const RemoveButton = styled.button.attrs({ type: "button" })`
  padding: 0;
  font-weight: bold;
  font-size: 32px;
  background-color: transparent;
  border: 0;
  margin: 10px 10px 10px auto;
  visibility: visibility;
  cursor: pointer;

  @media (min-width: 768px) {
    visibility: hidden;

    ${City}:hover & {
      visibility: visible;
    }
  }
`;

export const WeatherIcon = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  margin: auto 0;
`;

export const Stats = styled.div`
  margin: 10px;
  display: flex;
  align-items: center;
  justify-content: space-between;
`;

export const Stat = styled.h4`
  text-transform: capitalize;
  margin: 0;

  ${props =>
    props.major &&
    `
    font-size: 3rem;
  `};
`;
