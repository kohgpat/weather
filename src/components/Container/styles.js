import styled from "styled-components";

export const Container = styled.div`
  display: flex;
  margin: 0 auto;
  width: 100%;

  @media (min-width: 768px) {
    max-width: 1200px;
  }
`;
