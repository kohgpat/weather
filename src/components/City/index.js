import React from "react";
import PropTypes from "prop-types";
import WeatherIcon from "../WeatherIcon";
import WeatherDegree from "../WeatherDegree";
import * as s from "./styles";

const formatTemperature = temperature => {
  if (!temperature) {
    return null;
  }

  const temperatureString = temperature + "";

  if (temperatureString.split(".").length) {
    return temperatureString.split(".")[0];
  } else {
    return temperatureString;
  }
};

const formatInterval = interval => {
  if (!interval) {
    return null;
  }

  const millisecondsInSecond = 1000;
  const secondsInMinute = 60;

  return interval / millisecondsInSecond / secondsInMinute;
};

const City = ({ city, settings, onRemoveCity }) => {
  if (!city) {
    return null;
  }

  return (
    <s.City>
      <s.Header>
        <s.Name title={city.name}>{city.name}</s.Name>
        <s.RemoveButton onClick={() => onRemoveCity(city)}>
          &times;
        </s.RemoveButton>
      </s.Header>

      {city.params && (
        <s.SubHeader>
          <s.SubHeaderText>
            Update every {formatInterval(city.params.interval)} minutes
          </s.SubHeaderText>
        </s.SubHeader>
      )}

      {city.weather &&
        city.weather.length > 0 && (
          <s.WeatherMain>
            <s.Stat major>
              {city.weather && city.weather.length > 0 && city.weather[0].main}
            </s.Stat>
          </s.WeatherMain>
        )}

      <s.Stats>
        <s.Stat major>
          {city.main && formatTemperature(city.main.temp)}{" "}
          <WeatherDegree units={settings.units} />
        </s.Stat>

        {city.weather &&
          city.weather.length > 0 && (
            <s.Stat>
              <WeatherIcon weather={city.weather[0]} />
            </s.Stat>
          )}
      </s.Stats>
    </s.City>
  );
};

City.propTypes = {
  city: PropTypes.object.isRequired,
  settings: PropTypes.object.isRequired,
  onRemoveCity: PropTypes.func.isRequired
};

City.defaultProps = {
  cities: null,
  settings: null,
  onRemoveCity: null
};

export default City;
