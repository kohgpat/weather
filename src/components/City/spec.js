import React from "react";
import City from "./index";
import * as styles from "./styles";
import { shallow, mount } from "enzyme";

const city = {
  id: 1,
  name: "Moscow",
  weather: [{ main: { temp: "-10", description: "mist" } }]
};

const settings = {
  units: "celcius"
};

const onRemoveCity = jest.fn();

it("renders", () => {
  const component = shallow(
    <City city={{}} settings={settings} onRemoveCity={onRemoveCity} />
  );
  expect(component).toMatchSnapshot();

  const withCity = shallow(
    <City city={city} settings={settings} onRemoveCity={onRemoveCity} />
  );
  expect(withCity).toMatchSnapshot();
});

// TODO: jsdom currently not working properly with @supports css rule
// it("renders - city info", () => {
//   const component = mount(
//     <City city={city} settings={settings} onRemoveCity={onRemoveCity} />
//   );
//   expect(component.text()).toContain("Moscow");
// });

it("correctly invokes onRemoveCity callback", () => {
  const component = shallow(
    <City city={city} settings={settings} onRemoveCity={onRemoveCity} />
  );
  const removeButton = component.find(styles.RemoveButton);
  removeButton.simulate("click");
  expect(onRemoveCity).toBeCalled();
});
