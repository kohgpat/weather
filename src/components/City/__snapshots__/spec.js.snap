// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<styled.div>
  <styled.header>
    <styled.h1 />
    <styled.button
      onClick={[Function]}
    >
      ×
    </styled.button>
  </styled.header>
  <styled.div>
    <styled.h4
      major={true}
    >
       
      <WeatherDegree
        units="celcius"
      />
    </styled.h4>
  </styled.div>
</styled.div>
`;

exports[`renders 2`] = `
<styled.div>
  <styled.header>
    <styled.h1
      title="Moscow"
    >
      Moscow
    </styled.h1>
    <styled.button
      onClick={[Function]}
    >
      ×
    </styled.button>
  </styled.header>
  <styled.header>
    <styled.h4>
      Update every 
      0.0016666666666666668
       minutes
    </styled.h4>
  </styled.header>
  <styled.div>
    <styled.h4
      major={true}
    />
  </styled.div>
  <styled.div>
    <styled.h4
      major={true}
    >
       
      <WeatherDegree
        units="celcius"
      />
    </styled.h4>
    <styled.h4>
      <WeatherIcon
        weather={
          Object {
            "main": Object {
              "description": "mist",
              "temp": "-10",
            },
          }
        }
      />
    </styled.h4>
  </styled.div>
</styled.div>
`;
